plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/DefaultThriftMetastoreClientFactory.java

```diff
@@ -56,33 +55,32 @@
     private final int timeoutMillis;
     private final HiveMetastoreAuthentication metastoreAuthentication;
     private final String hostname;
+    private final ThriftHiveMetastoreState state;
```

Comment:
move `metastoreSupportsDateStatistics` there too

Then change commit message to something like
```
Encapsulate metastore introspection state

The knowledge about supported metastore operations collected at runtime
is captured as a bunch of fields (mostly some `AtomicInteger` objects).
The commit introduces a new class to keep this information, so that its lifecycle
can be easier managed.
```